### PR TITLE
Ability to customize CCT animations

### PR DESCRIPTION
**Pull request by FeiChen-okta**
#### Description:
Add options to customize the exit and entrance
animations for chrome custom tabs

#### Testing details:
- [x]  Verified basic functionality of change
- [x]  Added tests 

##### RESOLVES: 
[OKTA-264066](https://oktainc.atlassian.net/browse/OKTA-264066)

#### Primary Reviewer(s):
@okta/android

### CODE REVIEW

File: library/src/main/java/com/okta/oidc/CustomTabOptions.java [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+    @ColorInt
+    private int mCustomTabColor = -1;
+    @AnimRes
+    private int mStartEnterResId = -1;
+    @AnimRes
+    private int mStartExitResId = -1;
+    @AnimRes
+    private int mEndEnterResId = -1;
+    @AnimRes
+    private int mEndExitResId = -1;
```
Comment (helenchen-okta) [inline library/src/main/java/com/okta/oidc/CustomTabOptions.java]: Per this StackOverflow comment seems like R.anim ids tend to be negative: https://stackoverflow.com/questions/48835664/forcing-a-positive-integer-to-be-generated-from-ids-xml

I see them as negative in OV as well.

Comment (FeiChen-okta) [reply library/src/main/java/com/okta/oidc/CustomTabOptions.java]: Fixed

Comment (kavithachandramohan-okta) [reply library/src/main/java/com/okta/oidc/CustomTabOptions.java]: Dont we have default behavior ? default color and animation ?

Comment (FeiChen-okta) [reply library/src/main/java/com/okta/oidc/CustomTabOptions.java]: The default is the Android system animations. The SDK does not provide any.

File: library/src/main/java/com/okta/oidc/OktaAuthenticationActivity.java [hunk 2]
```diff
@@ -202,8 +203,19 @@ protected String getBrowser() {
...
+        if (mCustomTabOptions != null) {
+            if (mCustomTabOptions.getCustomTabColor() != 0) {
+                intentBuilder.setToolbarColor(mCustomTabOptions.getCustomTabColor());
+            }
+            if (mCustomTabOptions.getStartExitResId() != 0) {
+                intentBuilder.setStartAnimations(this,
+                        mCustomTabOptions.getStartEnterResId(),
+                        mCustomTabOptions.getStartExitResId());
+            }
+            if (mCustomTabOptions.getEndEnterResId() != 0) {
```
Comment (helenchen-okta) [inline library/src/main/java/com/okta/oidc/OktaAuthenticationActivity.java]: Why are we checking the exit res id for start but enter res id for end?

Comment (FeiChen-okta) [reply library/src/main/java/com/okta/oidc/OktaAuthenticationActivity.java]: If developers called the animation methods they should have set a correct id based on  @AnimRes. The check is to see if it's been called so one will suffice. If they set a value to 0 it is allowed but should be caught by AnimRes annotation.

**Pull request merged**